tests/tests_integration/test_api/test_geospatial.py

```diff
@@ -0,0 +1,410 @@
+import random
+import sys
+import time
+import uuid
+
+import pytest
+from cognite.client import utils
+from cognite.client.exceptions import CogniteAPIError
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes.geospatial import (
+    CoordinateReferenceSystem,
+    Feature,
+    FeatureList,
+    FeatureType,
+    FeatureTypeUpdate,
+    OrderSpec,
+    PropertyAndSearchSpec,
+)
+
+COGNITE_CLIENT = CogniteClient(max_workers=1)
+
+# sdk integration tests run concurrently on 3 python versions so this makes the CI builds independent from each other
+FIXED_SRID = 121111 + sys.version_info.minor
+
+
+@pytest.fixture()
+def test_crs():
+    wkt = """GEOGCS["WGS 84",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,
+    AUTHORITY["EPSG","7030"]],AUTHORITY["EPSG","6326"]],
+    PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["degree",0.0174532925199433,
+    AUTHORITY["EPSG","9122"]],AUTHORITY["EPSG","4326"]]"""
+    proj_string = """+proj=longlat +a=6377276.345 +b=6356075.41314024 +no_defs"""
+    crs = COGNITE_CLIENT.geospatial.create_coordinate_reference_systems(
+        crs=CoordinateReferenceSystem(srid=FIXED_SRID, wkt=wkt, proj_string=proj_string)
+    )
+    yield crs[0]
+    COGNITE_CLIENT.geospatial.delete_coordinate_reference_systems(srids=[FIXED_SRID])
+
+
+@pytest.fixture(params=[True, False])
+def allow_crs_transformation(request):
+    yield request.param
+
+
+@pytest.fixture()
+def test_feature_type():
+    external_id = f"FT_{uuid.uuid4().hex[:10]}"
+    feature_type = COGNITE_CLIENT.geospatial.create_feature_types(
+        FeatureType(
+            external_id=external_id,
+            properties={
+                "position": {"type": "POINT", "srid": "4326", "optional": "true"},
+                "volume": {"type": "DOUBLE"},
+                "temperature": {"type": "DOUBLE"},
+                "pressure": {"type": "DOUBLE"},
+            },
+            search_spec={"vol_press_idx": {"properties": ["volume", "pressure"]}},
+        )
+    )
+    yield feature_type
+    COGNITE_CLIENT.geospatial.delete_feature_types(external_id=external_id)
```

Comment:
Example on how to use the cognite_client fixture:
```suggestion
def test_feature_type(cognite_client: CogniteClient):
    external_id = f"FT_{uuid.uuid4().hex[:10]}"
    feature_type = cognite_client.geospatial.create_feature_types(
        FeatureType(
            external_id=external_id,
            properties={
                "position": {"type": "POINT", "srid": "4326", "optional": "true"},
                "volume": {"type": "DOUBLE"},
                "temperature": {"type": "DOUBLE"},
                "pressure": {"type": "DOUBLE"},
            },
            search_spec={"vol_press_idx": {"properties": ["volume", "pressure"]}},
        )
    )
    yield feature_type
    cognite_client.geospatial.delete_feature_types(external_id=external_id)
```